lib/smart-contract-client.js

```diff
@@ -37,37 +39,31 @@ const contractABI = [
   }
 ]
 
-// Contract address on the Filecoin EVM
-const CONTRACT_ADDRESS = '0x14183aD016Ddc83D638425D6328009aa390339Ce'
-// Singleton instance of the smart contract client
-let smartContract
 // Initialize provider and contract
 async function getSmartContractClient (rpcUrl) {
-  const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
+  const fetchRequest = new ethers.FetchRequest(rpcUrl)
+  fetchRequest.setHeader('Authorization', `Bearer ${RPC_AUTH}`)
```

Comment:
I find it weirdly inconsistent that `rpcUrl` is configured via function arguments and `RPC_AUTH` is hard-coded as a constant. 

Let's use the same mechanism for both.

Personally, I would simply use `RPC_URL` and `RPC_AUTH`.